crates/cairo-addons/src/vm/hint_definitions/dict.rs

```diff
@@ -0,0 +1,69 @@
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::{
+            dict_manager::DictManager,
+            hint_utils::{get_ptr_from_var_name, insert_value_from_var_name, insert_value_into_ap},
+        },
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::exec_scope::ExecutionScopes,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+
+use crate::vm::hints::Hint;
+
+pub const HINTS: &[fn() -> Hint] = &[dict_new_empty, copy_dict_segment];
+
+pub fn dict_new_empty() -> Hint {
```

Comment:
we could probably have a macro similar to the decorator in python to manage hint registration, but let's keep it like that for now

Reply:
I spent half-an-hour trying to make a macro, as an attribute `#[register_hint]` macro, but had some issues getting it to work properly. Thought this was an acceptable workaround